core-services/src/main/java/org/zalando/nakadi/service/publishing/BinaryEventPublisher.java

```diff
@@ -82,6 +82,9 @@ private List<NakadiRecordResult> processInternal(final EventType eventType,
                                                      final List<NakadiRecord> records,
                                                      final List<Check> checks,
                                                      final Map<HeaderTag, String> consumerTags) {
+
+        TracingService.setTag("number_of_events", String.valueOf(records.size()));
```

Comment:
There is `.setTag(String, Number)` method on `Span` - this might result in less memory footprint to store this tag data.

```suggestion
        TracingService.getActiveSpan().setTag("number_of_events", records.size());
```

Reply:
Good point.  Actually, the value has to be numeric for users to be able to use it in a query later.  I just overlooked the Span's methods.